### PR TITLE
Retrylater should renew the lock

### PR DESCRIPTION
**Pull request by xhl873**
draft pr for possible solution to renewlock in retrylater

### CODE REVIEW

File: message/retrylater.go [hunk 1]
```diff
@@ -20,11 +20,12 @@ type retryLaterHandler struct {
 	retryAfter time.Duration
 }
 
-func (r *retryLaterHandler) Do(ctx context.Context, orig Handler, message *servicebus.Message) Handler {
+func (r *retryLaterHandler) Do(ctx context.Context, orig Handler, message *servicebus.Message, sub *servicebus.Subscription) Handler {
 	select {
 	case <-ctx.Done():
 		return Error(fmt.Errorf("aborting retrylater for message. abandoning meddage %s: %w", message.ID, ctx.Err()))
 	case <-time.After(r.retryAfter):
```
Comment (paulgmiller) [inline message/retrylater.go:27]: if retryafter is longer than lockduration you have a problem. Maybe add a third case and renew after either peek lock or some arbitrary const for now

File: message/retrylater.go [hunk 2]
```diff
@@ -20,11 +20,12 @@ type retryLaterHandler struct {
...
+		sub.RenewLocks(ctx, message)
```
Comment (paulgmiller) [inline message/retrylater.go:28]: does renew locks have a return code?

**Comment by serbrech** [comment]
I think I prefer Paul's attempt that moves this higher up in the message handling code.
the Lock should always be renewed until the message is : 
- abandoned
- completed
- an error occurs
- we've reached a processing timeout or a maxlock timeout

thoughts?

**Comment by xhl873** [comment]
> I think I prefer @paulgmiller attempt that moves this higher up in the message handling code.
> the Lock should always be renewed until the message is :
> 
> * abandoned
> * completed
> * an error occurs
> * we've reached a processing timeout or a maxlock timeout
> 
> thoughts?

Yes, totally agree with you. We were exploring the possibilities at that time, and when I test this solution, found out that some of the msgs already expired before land on the handler. You are right, I think's the lock renew should be in higher levels

**Comment by serbrech** [comment]
closing now that #16 is merged

**Pull request closed without merging**